### PR TITLE
Server implemented PUT request

### PR DESCRIPTION
**Pull request by ABotnari**

### CODE REVIEW

File: Server/src/main/java/com/assistant/server/application/DummyController.java [hunk 1]
```diff
@@ -4,29 +4,28 @@
...
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.*;
 
-@Controller
+@RestController
 @RequiredArgsConstructor
-@RequestMapping("/dummy")
+@RequestMapping(value = "/dummy")
```
Comment (AbgarSim) [inline Server/src/main/java/com/assistant/server/application/DummyController.java:11]: value redundant here

File: Server/src/main/java/com/assistant/server/application/DummyController.java [hunk 2]
```diff
@@ -4,29 +4,28 @@
...
 public class DummyController {
 
 	private final DummyService service;
 
 	@GetMapping
-	public ResponseEntity<Long> getDummyLong() {
+	public ResponseEntity<Dummy> getDummyLong() {
```
Comment (AbgarSim) [inline Server/src/main/java/com/assistant/server/application/DummyController.java:17]: rename method to `getDummy()` pls

File: Server/src/main/java/com/assistant/server/infrastructure/entity/Dummy.java [hunk 3]
```diff
@@ -2,9 +2,7 @@
 
 import lombok.Data;
 
-import javax.persistence.Entity;
-import javax.persistence.Id;
-import javax.persistence.Table;
+import javax.persistence.*;
```
Comment (AbgarSim) [inline Server/src/main/java/com/assistant/server/infrastructure/entity/Dummy.java:5]: don't use wildcard imports - bad practice, revert this

File: Server/src/main/resources/application.properties [hunk 4]
```diff
@@ -3,3 +3,7 @@ spring.datasource.driver-class-name=org.postgresql.Driver
 spring.datasource.username=postgres
 spring.datasource.password=postgres
 spring.datasource.url=jdbc:postgresql://localhost:5432/assistant?autoreconnect=true
+
+spring.jpa.database-platform=org.hibernate.dialect.PostgreSQLDialect
+spring.jpa.show-sql=false
+spring.jpa.hibernate.ddl-auto=validate
```
Comment (AbgarSim) [inline Server/src/main/resources/application.properties:9]: newline at the end of file

**Review by AbgarSim — CHANGES_REQUESTED** [review]
plz fix

File: Server/src/main/java/com/assistant/server/application/DummyController.java [hunk 5]
```diff
@@ -4,29 +4,28 @@
...
-		if (service.exists(dummy))
-			return ResponseEntity.badRequest().body("Dummy already exists!");
-		String msg = service.saveDummy(dummy) + " Successfully created!";
-		return ResponseEntity.ok(msg);
+		return ResponseEntity.ok(service.saveDummy(dummy));
+	}
+
+	@PutMapping(value = "/update/{id}")
+	public ResponseEntity<Object> updateDummy(@RequestBody Dummy dummy, @PathVariable("id") Long id) {
+		return ResponseEntity.ok(service.updateDummy(id,dummy));
```
Comment (eugene-tech) [inline Server/src/main/java/com/assistant/server/application/DummyController.java:28]: Please use auto format

File: Server/src/main/java/com/assistant/server/service/DummyService.java [hunk 6]
```diff
@@ -11,15 +11,17 @@
 
 	private final DummyRepository repository;
 
-	public Long getDummyNumber() {
-		return repository.findFirstBy().getNum();
-	}
-
-	public boolean exists(Dummy dummy) {
-		return repository.existsById(dummy.getId());
+	public Dummy getDummyNumber() {
```
Comment (eugene-tech) [inline Server/src/main/java/com/assistant/server/service/DummyService.java:14]: Rename to getDummy

File: Server/src/main/java/com/assistant/server/application/DummyController.java [hunk 7]
```diff
@@ -4,29 +4,28 @@
...
 	@PostMapping("/add")
 	public ResponseEntity<Object> addDummy(@RequestBody Dummy dummy) {
-		if (service.exists(dummy))
-			return ResponseEntity.badRequest().body("Dummy already exists!");
-		String msg = service.saveDummy(dummy) + " Successfully created!";
-		return ResponseEntity.ok(msg);
+		return ResponseEntity.ok(service.saveDummy(dummy));
+	}
+
+	@PutMapping(value = "/update/{id}")
```
Comment (eugene-tech) [inline Server/src/main/java/com/assistant/server/application/DummyController.java:26]: Not sure if we need 'value='

**Review by eugene-tech — CHANGES_REQUESTED** [review]

File: Server/src/main/java/com/assistant/server/application/DummyController.java (hunk 1 above)
Comment (AlexandrBotnari) [reply Server/src/main/java/com/assistant/server/application/DummyController.java:11]: fixed

File: Server/src/main/java/com/assistant/server/application/DummyController.java (hunk 2 above)
Comment (AlexandrBotnari) [reply Server/src/main/java/com/assistant/server/application/DummyController.java:17]: done

File: Server/src/main/java/com/assistant/server/infrastructure/entity/Dummy.java (hunk 3 above)
Comment (AlexandrBotnari) [reply Server/src/main/java/com/assistant/server/infrastructure/entity/Dummy.java:5]: refactored

File: Server/src/main/resources/application.properties (hunk 4 above)
Comment (AlexandrBotnari) [reply Server/src/main/resources/application.properties:9]: added

**Comment by AbgarSim** [comment]
LGTM!

File: Server/src/main/java/com/assistant/server/service/DummyService.java [hunk 8]
```diff
@@ -11,15 +11,27 @@
...
+
+	public Dummy getDummy() {
+		return repository.findFirstBy();
 	}
 
-	public boolean exists(Dummy dummy) {
-		return repository.existsById(dummy.getId());
+	public DummyDTO saveDummy(DummyDTO dummyDTO) {
+		return dummyConverter.convertToDto(
+			repository.save(
```
Comment (eugene-tech) [inline Server/src/main/java/com/assistant/server/service/DummyService.java:22]: Lets inline this operation

File: Server/src/main/java/com/assistant/server/service/DummyService.java [hunk 9]
```diff
@@ -11,15 +11,27 @@
...
+			)
+		);
 	}
 
-	public Dummy saveDummy(Dummy dummy) {
-		return repository.save(dummy);
+	public DummyDTO updateDummy(final Long id, final DummyDTO dummyDto) {
+
+		final Dummy dummyById = repository.findDummyById(id);
+
```
Comment (eugene-tech) [inline Server/src/main/java/com/assistant/server/service/DummyService.java:31]: Why do we need blank lines here

File: Server/src/main/java/com/assistant/server/service/DummyService.java [hunk 10]
```diff
@@ -11,15 +11,27 @@
...
+				dummyConverter.convertToEntity(dummyDTO)
+			)
+		);
 	}
 
-	public Dummy saveDummy(Dummy dummy) {
-		return repository.save(dummy);
+	public DummyDTO updateDummy(final Long id, final DummyDTO dummyDto) {
+
+		final Dummy dummyById = repository.findDummyById(id);
```
Comment (eugene-tech) [inline Server/src/main/java/com/assistant/server/service/DummyService.java:30]: Why its final

**Pull request merged**